src/flash.c

```diff
@@ -73,11 +73,17 @@ bool writeEEPROM(bool blink)
   {
     FLASH_ClearFlag(FLASH_FLAG_EOP | FLASH_FLAG_PGERR | FLASH_FLAG_WRPRTERR);
 
-    FLASH_ErasePage(FLASH_WRITE_ADDR);
-    status = FLASH_ErasePage(FLASH_WRITE_ADDR + FLASH_PAGE_SIZE);
+    for(int i = 0; i < NUM_PAGES; i++)
+      status = FLASH_ErasePage(FLASH_WRITE_ADDR + i * FLASH_PAGE_SIZE);
     for (unsigned int i = 0; i < sizeof(params_t) && status == FLASH_COMPLETE; i += 4)
+    {
       status = FLASH_ProgramWord(FLASH_WRITE_ADDR + i, *(uint32_t *)((char *)&_params + i));
-    if (status == FLASH_COMPLETE)
+      if (status != FLASH_COMPLETE)
+      {
+        continue;
+      }
```

Comment:
Yeah, I don't really know why I did that. You're probably right.  I'll figure out what is going on here and do it right.